static/js/util/test_utils.js

```diff
@@ -21,14 +22,38 @@ export function findCourse(courseSelector: (course: Course, program: Program) =>
   throw "Unable to find course";
 }
 
-export const alterFirstRun = (course: Course, overrideObject: Object): Course => {
+export const alterFirstRun = (course: Course, overrideObject: Object): CourseRun => {
   course.runs[0] = Object.assign({}, course.runs[0], overrideObject);
+  return course.runs[0];
 };
 
 export function findAndCloneCourse(courseSelector: (course: Course, program: Program) => boolean): Course {
   return _.cloneDeep(findCourse(courseSelector));
 }
 
+export function setCourseRunCount(course: Course, desiredRunCount: number, courseRunToCopy: ?CourseRun) {
+  let currentRunCount = course.runs.length;
+  if (currentRunCount < desiredRunCount) {
+    let courseRun = currentRunCount === 0 ? courseRunToCopy : course.runs[0];
+    if (!courseRun) {
+      throw new Error('Need a course run to copy.');
+    }
+    let runsNeeded = desiredRunCount - currentRunCount;
+    let idMax = _.max(_.map(course.runs, run => run.id)) || 0;
```

Comment:
`course_id` and `id` need to be unique within the response data, not just within the course or program. If we were not going to check `DASHBOARD_RESPONSE` or other response objects then this function would need to create a new response object, not just a new course